desk/app/chat.hoon

```diff
@@ -1528,11 +1637,69 @@
       %-  (slog leaf/"Failed to notify about dm {<ship>}" u.p.sign)
       di-core
     ::
-        [%proxy ~]
+        [%proxy *]
       ?>  ?=(%poke-ack -.sign)
       ?~  p.sign  di-core
-      ::  TODO: handle?
-      %-  (slog leaf/"Failed to dm {<ship>}" u.p.sign)
+      ::  if we already tried hard, this is the end of the road.
+      ::
+      ?:  ?=([%archaic ~] t.wire)
+        %-  (slog leaf/"Failed to dm {<ship>} again" u.p.sign)
+        di-core
+      ::  if they're just known version mismatching, we should do nothing
+      ::TODO  might want to mark the local msg as undelivered though...
+      ::
+      ?.  ?=(%await (read-status:neg bowl ship dap.bowl))
+        %-  (slog leaf/"Failed to dm {<ship>}" u.p.sign)
+        di-core
+      ::  if a poke failed, but we also don't have a negotiated version for
+      ::  them, they might still be on the old (pre-2024) chat backend. try
+      ::  sending them an old-style poke if we can, for backcompat.
+      ::  we do our best to recover the message from the wire.
+      ::
+      =.  cor
+        =;  c=(unit cage)
+          ?~  c  cor
+          %+  emit  %pass
+          [(weld di-area /proxy/archaic) %agent [ship %chat] %poke u.c]
+        ?+  t.wire  ~
+            [%rsvp @ ~]
+          =/  ok=?  ;;(? (slav %f i.t.t.wire))
+          `[%dm-rsvp !>(`rsvp:dm:old`[our.bowl ok])]
+        ::
+            [@ ~]
+          %-  some
+          :-  %dm-diff
+          !>  ^-  diff:dm:old
+          =/  id=time  (slav %ud i.t.wire)
+          :-  [our.bowl id]
+          =/  msg=(unit [=time =writ:c])
+            (get:di-pact our.bowl id)
```

Comment:
Because it's a message _we_ are trying to send, the "full" id will always start with `our`, correct?

Reply:
yep